Source/OCMockTests/OCMockObjectMacroTests.m

```diff
@@ -291,6 +291,28 @@ - (void)testSetsUpReject
     XCTAssertThrows([mock verify], @"Should have complained about rejected method being invoked");
 }
 
+- (void)testDontAllowAndBlocksOnRejects
+{  
+    NSNotification *notification = [NSNotification notificationWithName:@"Notification" object:nil];
+
+    // Version 3 syntax.
+    id mock = OCMClassMock([TestClassForMacroTesting class]);
+    XCTAssertThrows(OCMReject([mock stringValue]).andPost(notification));
+
+    // Version 2 syntax.
+    mock = OCMClassMock([TestClassForMacroTesting class]);
+    XCTAssertThrows([[[mock reject] andPost:notification] stringValue]);
+
+    // Mixes of version 2 and version 3 syntax.
+    mock = OCMClassMock([TestClassForMacroTesting class]);
+    XCTAssertThrows(OCMExpect([[[mock andPost:notification] never] stringValue]));
```

Comment:
I'm going to merge this without this particular test. That's a complete misuse of the API and, if anything, a different error should be shown. In fact if we started to try to detect invalid sequences of method invocations in the fluent API, that would be a major effort. Not including this case makes the implementation simpler; the change in `setMatchAndReject:` is only required for this case.